src/main/php/web/session/CookieBased.class.php

```diff
@@ -0,0 +1,165 @@
+<?php namespace web\session;
+
+use lang\FormatException;
+use util\Secret;
+use web\Cookie;
+use web\session\cookie\{Session, Format};
+
+class CookieBased extends Sessions {
+  private $key, $format;
+  private $attributes= [
+    'path'     => '/',
+    'secure'   => true,
+    'domain'   => null,
+    'httpOnly' => true,
+    'sameSite' => 'Lax'
+  ];
+
+  /**
+   * Creates an new cookie-based session
+   *
+   * @param  string|util.Secret $key
+   * @param  ?web.session.cookie.Format $format
+   */
+  public function __construct($key, Format $format= null) {
+    $this->key= $key instanceof Secret ? $key : new Secret($key);
+    $this->format= $format ?? Format::available()[0];
+  }
+
+  /**
+   * Sets the session transport
+   *
+   * @param  web.session.Transport $transport
+   * @return self
+   */
+  public function via($transport) {
+
+    // Special-case handling for Cookies-transport
+    if ($transport instanceof Cookies) {
```

Comment:
Should the transport really be abstracted from the session implemenation? Will we use any other transport than cookies?

Reply:
This should be done in another refactoring-only PR.

Reply:
https://github.com/xp-forge/sessions/pull/11